### DISCUSSION TITLE
FindTransferAnchors: how do column sums of weights matrix all equal 1?

### DISCUSSION THREAD

**Discussion by danielcgingerich** [stuart-lab/signac#658] [Q&A] [answered]
Hi, 

I am just wondering how this was figured out.  From anchor weighting I see that:
D = [1 - dist(c, a)/dist(c, a.kweight) ]*anchor.score 
gaussian kernal applied to D
D = 1-exp(-D/(2/sd)^2) 
and then normalized by all surrounding anchors for the anchor weight.  

How is it that the resulting anchors sum to 1?

**Comment by timoast** [comment, marked as answer]
We make them sum to 1 by dividing by the column sums: https://github.com/satijalab/seurat/blob/master/src/integration.cpp#L80
